server/shared/src/main/scala/org/http4s/server/middleware/EntityLimiter.scala

```diff
@@ -50,8 +50,15 @@ object EntityLimiter {
     _.pull
       .take(n)
       .flatMap {
-        case Some(rest) => // if rest is empty then we won't raise an error
-          (rest >> Stream.raiseError(EntityTooLarge(n))).pull.echo
+        case Some(rest) =>
+          // Drain the rest of the entity. If `rest` is empty then we won't raise an error
+          rest.last
```

Comment:
It will increase the latency for no apparent benefit, IMHO. We should eagerly respond with 413 oppositely.

Reply:
https://github.com/http4s/http4s/pull/7603
It's a slightly different approach to the problem, but I was generally leaning towards dropping that PR. Sorry for jumping ahead of you!